sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/BiometricAuthWithoutCryptoCheck.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2018-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.psiUtil.getCallNameExpression
+import org.jetbrains.kotlin.psi.psiUtil.isNull
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.AbstractCheck
+import org.sonarsource.kotlin.api.FunMatcher
+import org.sonarsource.kotlin.api.SecondaryLocation
+import org.sonarsource.kotlin.api.matches
+import org.sonarsource.kotlin.converter.KotlinTextRanges.textRange
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+private val AUTHENTICATE_FUN_MATCHERS = listOf(
+    FunMatcher {
+        withNames("authenticate")
+        qualifier = "android.hardware.biometrics.BiometricPrompt"
+    },
+    FunMatcher {
```

Comment:
You can use constructor arguments instead of lambda here. No big preference here, but to my taste constructor looks more concise